File: Practice/3626-smallest-divisible-digit-product-i/smallest-divisible-digit-product-i.py
class Solution:
    def productOfDigits(self, num: int) -> int:
        ans = 1
        while num>0:
            temp = num%10
            ans = ans*temp
            num = num//10

        return ans


    def smallestNumber(self, n: int, t: int) -> int:
        for i in range(n, 101):
            temp = self.productOfDigits(i)
            if temp%t == 0:
                return i

        return -1

        

File: POTD/3626-smallest-divisible-digit-product-i/smallest-divisible-digit-product-i.cpp
class Solution {
private:
    int productOfDigits(int num){
        int ans = 1;
        while(num){
            int digit = num%10;
            ans = ans*digit;
            num=num/10;
        }

        return ans;
    }
    
public:
    int smallestNumber(int n, int t) {
        for(int i=n; i<=100; i++){
            int prod = productOfDigits(i);
            cout << prod;
            if(prod%t == 0){
                return i;
            }
        }

        return -1;
    }
};